src/components/review.jsx

```diff
@@ -1,125 +1,186 @@
-  };
-
-  const header = [
-    <ReviewBlurb key='review-blurb'
-      card={card}
-      primaryRepoName={primaryRepoName} />,
-  ];
-
-  let reactionsStat = {
-    THUMBS_UP: 0,
-    THUMBS_DOWN: 0,
-    LAUGH: 0,
-    HOORAY: 0,
-    HEART: 0,
-    CONFUSED: 0
-  };
-  if (reactions) {
-    reactions.forEach(reaction => reactionsStat[reaction.content]++);
-  }
+      <div className='-card-and-related'>
+        <BS.ListGroupItem
+          key={key}
+          header={header}
+          className={classnames(classes)}>
 
-  return (
-    <div className='-card-and-related'>
-      <BS.ListGroupItem
-        key={key}
-        header={header}
-        className={classnames(classes)}>
-
-        <span className='-extra-span-for-inline-popover'>
-          <a
-            key='link'
-            className='review-title'
-            target='_blank'
-            href={url}>
-              <GithubFlavoredMarkdown
-                inline
-                repoOwner={repoOwner}
-                repoName={repoName}
-                text={bodyText}/>
-          </a>
-        </span>
-
-        <span key='footer' className='review-footer'>
-          <span key='left-footer' className='comment-reactions'>
-            <Reactions stat={reactionsStat}/>
+          <span className='-extra-span-for-inline-popover'>
+            <a
+              key='link'
+              className='review-title'
+              target='_blank'
+              href={url}>
+                <GithubFlavoredMarkdown
+                  inline
+                  repoOwner={repoOwner}
+                  repoName={repoName}
+                  text={bodyText}/>
+            </a>
           </span>
-          <span key='right-footer' className='review-time-and-user'>
-            <Time key='time' className='updated-at' dateTime={updatedAt}/>
-            {assignedAvatar}
+
+          <span key='footer' className='review-footer'>
+            <span key='left-footer' className='comment-reactions'>
+              <Reactions
+                id={id}
+                stat={reactionsStat}
+                noReactionByMe={noReactionByMe}
+                hasLogin={hasLogin}
+                saveCallBack={this.saveToDatabase}/>
+            </span>
+            <span key='right-footer' className='review-time-and-user'>
+              <Time key='time' className='updated-at' dateTime={updatedAt}/>
+              {assignedAvatar}
+            </span>
           </span>
-        </span>
-      </BS.ListGroupItem>
-      <div key='related' className='related-issues'>
-        {relatedCards}
+        </BS.ListGroupItem>
+        <div key='related' className='related-issues'>
+          {relatedCards}
+        </div>
       </div>
-    </div>
-  );
+    );
+  }
 }
 
-function Review({review}) {
+function Review({review, loginInfo}) {
   return (
-    <ReviewCard card={review}/>
+    <ReviewCard card={review} loginInfo={loginInfo}/>
   );
 }
 
-export default Review;
+export default withAuth(Review);
```

Comment:
withAuth() should be used for the existing components which need it, i.e. dashboard and kanban board where you added it for the filters.

Reply:
Use component with render prop looks nice, but I want to reuse the onChangeToken logic. I changed dashboard logic a bit, so it's not a problem anymore.